lib/openhab/core/dsl/items/datetime_item.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'forwardable'
+require 'java'
+require 'time'
+require 'openhab/core/dsl/types/datetime'
+
+module OpenHAB
+  module Core
+    module DSL
+      module Items
+        #
+        # Delegation to OpenHAB Number Item
+        #
+        class DateTimeItem
+          extend Forwardable
+
+          def_delegator :@datetime_item, :to_s
+
+          def initialize(datetime_item)
+            @datetime_item = datetime_item
+            super()
+          end
+
+          def to_dt
+            OpenHAB::Core::DSL::Types::DateTime.new(@datetime_item.state)
+          end
+
+          def <=>(other)
+            logger.trace("Comparing self (#{self.class}) to #{other} (#{other.class})")
+            case other
+            when DateTimeItem
+              to_dt <=> other.to_dt
+            else
+              to_dt <=> other
+            end
+          end
+
+          def <(other)
+            (self <=> other).negative?
+          end
+
+          def <=(other)
+            (self <=> other) <= 0
+          end
+
+          def >(other)
+            (self <=> other).positive?
+          end
+
+          def >=(other)
+            (self <=> other) >= 0
+          end
+
+          def ==(other)
+            (self <=> other).zero?
+          end
+
+          def zone
+            to_dt.zone
+          end
+
+          def method_missing(meth, *args, &block)
```

Comment:
For datetime can we SimpleDelegator rather than method missing? I don't see any special logic in here beyond what SimpleDelegator gives us.

Reply:
Haven't used it much before, but you're probably right, will read up a bit on it. But we want it to delegate to DateTimeItem first, what happens if there are name conflicts? Doesn't the delegator take precedence?

Reply:
I tried to use SimpleDelegator, but still had to implement `method_missing` to be able to delegate to the DateTimeType object kept as the state, and the code looked uglier:
```ruby
def method_missing(meth, *args, &block)
  super
rescue NoMethodError
  if state?
    to_dt.send(meth, *args, &block)
  end
end
```

Reply:
Ok, so it is really doing ordered multi-delegation?  Delegates to A, if A doesn't have the method, delegates to B, etc?

Reply:
Exactly, first it checks if the real OpenHAB item has the method, otherwise it delegates to a DateTime object that wraps the state, so it can be accessed without having to write `Item.state` everywhere.